RFS/src/main/java/com/rfs/cms/IWorkCoordinator.java

```diff
@@ -0,0 +1,80 @@
+package com.rfs.cms;
+
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.time.Instant;
+
+public interface IWorkCoordinator extends AutoCloseable {
```

Comment:
I do not yet grok this interface, what do you think about adding readme /w diagram?

Reply:
So there is the problem of allocating work, then there is the problem of defining work to the allocated, having a single interface in one place makes this harder to understand, seems like this would be better suited to be in two interfaces - in the case where there is a dual consumer and producer, it could just implement both interfaces.  What do you think?

Reply:
I've added a lot more documentation & pulled the core members to the top to improve readability.  I'll try to get Chris's original documentation on this algorithm into the tree.